### ISSUE TITLE
`useDeepCompareEffect` should not show the warning when the values are not primitive but `null`/`undefined` at the moment

### ISSUE THREAD

**Issue by italodeandra** [streamich/react-use#2227]
I have this code that uses `useDeepCompareEffect` and when I have on the deps a value that is not a primitive value but at the moment it is `null` or `undefined`, it shows the following message:

```
`useDeepCompareEffect` should not be used with dependencies that are all primitive values. Use React.useEffect instead.
```

In my opinion this message should be removed for `null` and `undefined` values.

**Comment by HummingBird24** [comment]
Super late but ```useDeepCompareEffectNoCheck``` was created for this use case.

From the README: 

```NOTE: Only use this if your values are objects or arrays that contain objects. Otherwise, you should just use React.useEffect. In case of "polymorphic" values (eg: sometimes object, sometimes a boolean), use useDeepCompareEffectNoCheck, but do it at your own risk, as maybe there can be better approaches to the problem.```

**Comment by HummingBird24** [comment]
@italodeandra I agree with you actually.  It's normal to expect an object variable to possibly be  ```null``` or ```undefined``` but not a ```boolean```. So yeah although ```useDeepCompareEffectNoCheck``` gets around the problem, I don't think it should be required. It also confuses things with the naming.

**Comment by xixixao** [comment]
useDeepCompareEffectNoCheck doesn't exist anymore and useCustomCompareEffect has the same code inside, so there's no way to get around this besides rolling your own implementation.

**Comment by thomasmoon** [comment]
> Super late but `useDeepCompareEffectNoCheck` was created for this use case.
> 
> From the README:
> 
> `NOTE: Only use this if your values are objects or arrays that contain objects. Otherwise, you should just use React.useEffect. In case of "polymorphic" values (eg: sometimes object, sometimes a boolean), use useDeepCompareEffectNoCheck, but do it at your own risk, as maybe there can be better approaches to the problem.`

I have a polymorphic case with objects or plain strings. It works fine in practice, so it should be possible to disable the warning?

`useDeepCompareEffectNoCheck` seems to not exist now.
